File: src/models/ShopProduct.js
/*  ShopProduct = {
        "id": "3",
        "shopId": "3",
        "shopName": "",
        "productId": "1",
        "productName": "",
        "pricePerUnit": 11.30,
        "measuringUnits": "box"
    } */
class ShopProduct {
  constructor(
    idOrObj,
    shopId,
    productId,
    pricePerUnit,
    measuringUnits
    // ,shopName
    // ,productName
  ) {
    console.log("arguments.length=" + arguments.length + ", id: " + idOrObj + ", shopId: " + shopId + ", productId: " + productId);
    if (arguments.length === 1) {
      //  First and only argument is an object
      this.id = idOrObj.id;
      this.shopId = idOrObj.shopId;
      this.productId = idOrObj.productId;
      this.pricePerUnit = idOrObj.pricePerUnit;
      this.measuringUnits = idOrObj.measuringUnits;
      // this.shopName = idOrObj.shopName;
      // this.productName = idOrObj.productName;
    } else {
      this.id = idOrObj;
      this.shopId = shopId;
      this.productId = productId;
      this.pricePerUnit = pricePerUnit;
      this.measuringUnits = measuringUnits;
      // this.shopName = shopName;
      // this.productName = productName;
    }
  }
  toString() {
    return (
      '{ "id": "' +
      this.id +
      '", "shopId": "' +
      this.shopId +
      '", "shopName": "' +
      this.shopName +
      '", "productId": "' +
      this.productId +
      '", "productName": "' +
      this.productName +
      '", "pricePerUnit": ' +
      this.pricePerUnit +
      ', "measuringUnits": "' +
      this.measuringUnits +
      '" }'
    );
  }
}

File: price_compare_app/src/main/java/com/beniregev/price_compare_app/model/ShopProduct.java
package com.beniregev.price_compare_app.model;

import lombok.*;
import org.springframework.data.annotation.Id;

import java.math.BigDecimal;
import java.util.UUID;

@NoArgsConstructor
@EqualsAndHashCode
@ToString
public class ShopProduct {
    @Id
    @Getter
    private String id;
    @Setter @Getter
    private String shopId;
    @Setter @Getter
    private String shopName;
    @Setter @Getter
    private String productId;
    @Setter @Getter
    private String productName;
    @Setter @Getter
    private BigDecimal pricePerUnit;
    @Setter @Getter
    private String measuringUnits;

    public ShopProduct(String shopId, String productId, BigDecimal pricePerUnit, String measuringUnits) {
        this.id = UUID.randomUUID().toString();
        this.shopId = shopId;
        this.productId = productId;
        this.pricePerUnit = pricePerUnit;
        this.measuringUnits = measuringUnits;
        this.shopName = "";
        this.productName = "";
    }
}
